testng-core/src/test/java/test/reports/FailedReporterTest.java

```diff
@@ -81,4 +87,65 @@ public void testParameterPreservationWithFactory() throws IOException {
 
     assertThat(myDiff).matches((it) -> !it.hasDifferences(), "!it.hasDifferences()");
   }
+
+  @Test(dataProvider = "getTestData")
+  public void testToEnsureConfigFailuresAreIncluded(
+      Pair<Class<?>, String> pairA, Pair<Class<?>, String> pairB) throws IOException {
```

Comment:
Not sure that using `Pair` is more readable than 2 params well named.

Reply:
@juherr I need to be sending in 2 classes and 2 test method names. I had the choice of either creating a new Pojo for this and using that or using something that already exists. 

`Pair` sounded apt for me as well because it's a pair of class and a method.

Reply:
Sure `Pair` is the appropriate object to use but maybe a bit excessive in this specific case because `Class<?> testClassA` and `String testMethodA` are more expressive.

In some other languages, you are be able to do things like `(testClassA, testMethodA) = pairA`.

Reply:
@juherr You want me to add `testClassA, testMethodA, testClassB, testMethodB` as my data driven method's parameter?

Reply:
That was just an open suggestion. Please, check the initial issue first.